examples/nodejs/token-vending-machine/infrastructure/lib/token-vending-machine-stack.ts

```diff
@@ -46,6 +48,48 @@ export class TokenVendingMachineStack extends cdk.Stack {
       requestTemplates: {'application/json': '{ "statusCode": "200" }'},
     });
 
-    api.root.addMethod('GET', tvmIntegration);
+    if (config.authenticationMethod === "lambda-authorizer") {
```

Comment:
stylistic nit - when branching on an enum like this, i usually find a `switch` statement to be more readable. but it's subjective :)

in either case, whether you stick with an `if` or move to a `switch`, you should have a final `else` or `default` branch that throws an error if you get an unexpected value. That way if we add a new option to the enum in the future and forget to update the code here, we'll at least find out about it at runtime.